front/src/containers/studentPage.js

```diff
@@ -49,13 +50,96 @@ function StudentPage({exercice}) {
     pages: ['Exo 1', 'Exo 2', 'Exo 3', 'Exo 4'],
   };
   const {user, logoutUser} = useContext(AuthContext);
+
+  const [code, setCode] = useState('#enter your code here!');
+  const [listExercises, setListExercises] = useState([{statement: ''}]);
+  const [activeExerciseIndex, setActiveExerciseIndex] = useState(0);
+  const [activeExerciseStatement, setActiveExerciseStatement]= useState('');
+
+  const fetchExercises = async () => {
+    const response = await fetch(
+        'http://localhost:8000/exercise/',
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          credentials: 'include',
+          body: JSON.stringify({
+            'classroom': user.classroom,
+          }),
+        },
+    );
+    if (response.status === 200) {
+      const content = await response.json();
+      setListExercises(content);
+      console.log(listExercises[activeExerciseIndex]);
+    } else {
+      alert('Didn\'t work');
+    }
+  };
+
+  useEffect(() => {
+    fetchExercises();
+  }, []);
+
+  useEffect(() => {
+    setActiveExerciseStatement(listExercises[activeExerciseIndex].statement);
+  }, [activeExerciseIndex, listExercises]);
+
+  useEffect(() => {
+    const interval = setInterval(fetchExercises, 10000);
+    return () => clearInterval(interval);
+  }, [listExercises]);
+
+  const onChange = (action, data) => {
+    switch (action) {
+      case 'code': {
+        setCode(data);
+        console.log(typeof(code));
+        break;
+      }
+      default: {
+        console.warn('case not handled!', action, data);
+      }
+    }
+  };
+
+  const onClickSubmit = async (event) => {
+    const response = await fetch(
+        'http://localhost:8000/solution/create/',
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          credentials: 'include',
```

Comment:
Miss authentification, it is working because back is not well done but should have the access token and remove credentials.

Reply:
It's weird, I thought I have pushed the right version, because I changed that on my computer